silkrpc/http/request_handler.cpp

```diff
@@ -49,6 +77,59 @@ boost::asio::awaitable<void> RequestHandler::handle_request(const http::Request&
             co_return;
         }
 
+        if (jwt_secret_.length() > 0) {
+            const auto it = std::find_if(request.headers.begin(), request.headers.end(), [&](const Header& h){
+                        return h.name == "Authorization";
+                    });
+            if (it == request.headers.end()) {
+                SILKRPC_ERROR << "jwt request without Authorization in auth connection" << "\n";
+                reply.content = make_json_error(request_id, 403, "missing Authorization Header").dump() + "\n";
+                reply.status = http::Reply::unauthorized;
+                reply.headers.reserve(2);
+                reply.headers.emplace_back(http::Header{"Content-Length", std::to_string(reply.content.size())});
+                reply.headers.emplace_back(http::Header{"Content-Type", "application/json"});
+                SILKRPC_INFO << "handle_request t=" << clock_time::since(start) << "ns\n";
+                co_return;
+            }
+
+            std::string client_token;
+            if (it->value.substr(0, 7) == "Bearer ") {
+                client_token = it->value.substr(7);
+            } else {
+                SILKRPC_ERROR << "jwt client request without token: " << "\n";
+                reply.content = make_json_error(request_id, 403, "missing token").dump() + "\n";
+                reply.status = http::Reply::unauthorized;
+                reply.headers.reserve(2);
+                reply.headers.emplace_back(http::Header{"Content-Length", std::to_string(reply.content.size())});
+                reply.headers.emplace_back(http::Header{"Content-Type", "application/json"});
+                SILKRPC_INFO << "handle_request t=" << clock_time::since(start) << "ns\n";
+                co_return;
+            }
+
+            try {
+               // Parse token
+               auto decoded_client_token = jwt::decode(client_token);
+
+               // Validate token
+               auto verifier = jwt::verify()
+                   .allow_algorithm(jwt::algorithm::hs256{convert_jwt_secret(jwt_secret_)});
```

Comment:
I think `convert_jwt_secret` function is not needed: you can use `to_hex` and then convert it to `std::string` like this:
```
const auto jwt_secret_bytes = *from_hex(jwt_secret_); // must *not* fail (should check it)
auto verifier = jwt::verify()
    .allow_algorithm(jwt::algorithm::hs256{jwt_secret_bytes.cbegin(), jwt_secret_bytes.cend()});
```

Then, we can do better than this and avoid converting the JWT secret each time: just convert it once (checking the result) when generated/loaded in the `Daemon::run` and then just pass it as `const Bytes&` where needed.

So, basically all the classes have `const Bytes& jwt_secret_;` instead of `const std::string& jwt_secret_;`